update.sh

```diff
@@ -1,10 +1,23 @@
 #!/bin/bash
+
 INSTALL_DIR="/opt/zpui"
 SUDO=''
+
+cd "$(dirname "$0")"  # Make sure we are in the script's directory when running
+set -e  # Strict mode : script stops if any command fails
+
+
 if (( $EUID != 0 )); then
     SUDO='sudo'
 fi
-git pull
-$SUDO rsync -av --delete ./  $INSTALL_DIR
-$SUDO systemctl restart zpui.service
 
+git pull origin master  # Always tell the branch we pull from
+
+${SUDO} pip install -r requirement.txt  # Make sure we have the latest dependencies installed
+
+# Run tests
+pytest2 --doctest-modules -v --doctest-ignore-import-errors --ignore=apps/example_apps/fire_detector/ --ignore=ui/tests/test_checkbox.py  #  todo : fixes checkbox testing not working at the moment
+
+${SUDO} mkdir -p ${INSTALL_DIR}
```

Comment:
The directory is supposed to exist at this point, are you sure this is necessary, or even helpful? Like, the install directory is created by `setup.sh`, and rsync mkdir's the directory anyway, from my experience.

Reply:
This is just a nitpick, though, I'm mostly interested in your reasoning - maybe you've experienced some problem with it

Reply:
The reasoning is that it ensures the directory exists in two cases :
1. The name changed for any reason
2. The user installed it manually in the wrong dir

I recognize they are some far fetched edge-cases, but I thought "better safe than sorry". Since it's such a crucial process, I want it to be as safe as possible